src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs

```diff
@@ -9,24 +9,25 @@
 using System.Threading.Tasks;
 using Xunit;
 
-namespace System.Text.Json.Tests.Serialization
+namespace System.Text.Json.Serialization.Tests
 {
-    public static partial class CollectionTests
+#if !BUILDING_SOURCE_GENERATOR_TESTS
+    public abstract partial class CollectionTests
     {
         [Theory]
         [MemberData(nameof(GetAsyncEnumerableSources))]
-        public static async Task WriteRootLevelAsyncEnumerable<TElement>(IEnumerable<TElement> source, int delayInterval, int bufferSize)
+        public async Task WriteRootLevelAsyncEnumerable<TElement>(IEnumerable<TElement> source, int delayInterval, int bufferSize)
         {
             JsonSerializerOptions options = new JsonSerializerOptions
             {
                 DefaultBufferSize = bufferSize
             };
 
-            string expectedJson = JsonSerializer.Serialize(source);
+            string expectedJson = await JsonSerializerWrapperForString.SerializeWrapper(source);
```

Comment:
Since this method call is only used to generate the expected JSON string, we don't really need to exercise any async APIs here.